src/Sarif.Sarifer/Commands/SariferPackageCommand.cs

```diff
@@ -60,7 +60,8 @@ private static void GetFilesRecursive(List<string> targetFiles, ProjectItem proj
             }
             else
             {
-                string localPath = projectItem.Properties.Item("LocalPath").Value.ToString();
+                // handle some solution items has null property. e.g. Solution Items, avoid null reference exception
+                string localPath = projectItem.Properties?.Item("LocalPath").Value.ToString();
```

Comment:
what is the properties type?

Reply:
I think these are COM objects

projectItem.GetType().ToString()
"Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAProjectItem"
projectItem.Properties.GetType().ToString()
"Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAProperties"

---
In reply to: [603657849](https://github.com/microsoft/sarif-visualstudio-extension/pull/387#discussion_r603657849) [](ancestors = 603657849)